library/cwm/test/multi_status_selector_test.rb

```diff
@@ -0,0 +1,339 @@
+    end
+  end
+
+  describe "#auto_select!" do
+    it "sets item as auto selected" do
+      subject.auto_select!
+
+      expect(subject.status).to eq(:auto_selected)
+    end
+  end
+
+  describe "#to_s" do
+    it "returns a string" do
+      expect(subject.to_s).to be_a(String)
+    end
+
+    context "when the item is enabled" do
+      it "includes a link for the input" do
+        expect(subject.to_s).to match(regexp_input_link)
+      end
+
+      it "includes a link for the label" do
+        expect(subject.to_s).to match(regexp_label_link)
+      end
+    end
+
+    context "when the item is not enabled" do
+      let(:enabled) { false }
+
+      it "uses a grey color" do
+        expect(subject.to_s).to match(/.*color: grey.*/)
+      end
+
+      it "includes the item label" do
+        expect(subject.to_s).to include(subject.label)
+      end
+
+      it "does not include a link for the input" do
+        expect(subject.to_s).to_not match(regexp_input_link)
+      end
+
+      it "does not include a link for the label" do
+        expect(subject.to_s).to_not match(regexp_label_link)
+      end
+    end
+
+    context "when running in text mode" do
+      before { allow(Yast::UI).to receive(:TextMode).and_return(true) }
+
+      context "and the item is selected" do
+        let(:status) { :selected }
+
+        it "displays `[x]` as icon" do
+          expect(subject.to_s).to include("[x]")
+        end
+      end
+
+      context "and the item is auto selected" do
+        let(:status) { :auto_selected }
+
+        it "displays `[a]` as icon" do
+          expect(subject.to_s).to include("[a]")
+        end
+      end
+
+      context "and the item is not selected" do
+        let(:status) { :unselected }
+
+        it "displays `[ ]` as icon" do
+          expect(subject.to_s).to include("[ ]")
+        end
+      end
+
+      context "and the item has an unknown status" do
+        let(:status) { :unknown }
+
+        it "displays `[ ]` as icon" do
+          expect(subject.to_s).to include("[ ]")
+        end
+      end
+    end
+
+    context "when NOT running in text mode" do
+      before { allow(Yast::UI).to receive(:TextMode).and_return(false) }
+
+      context "and the item is selected" do
+        let(:status) { :selected }
+
+        it "displays the proper icon" do
+          expect(subject.to_s).to include("checkbox-on.svg")
+        end
+      end
+
+      context "and the item is auto selected" do
+        let(:status) { :auto_selected }
+
+        it "displays the proper icon" do
+          expect(subject.to_s).to include("auto-selected.svg")
+        end
+      end
+
+      context "and the item is not selected" do
+        let(:status) { :unselected }
+
+        it "displays the proper ion" do
+          expect(subject.to_s).to include("checkbox-off.svg")
+        end
+      end
+
+      context "and the item has an unknown status" do
+        let(:status) { :unknown }
+
+        it "displays the proper ion" do
```

Comment:
it is a bit confusing here. Why not mention that it shows unselected icon? and there is also typo "ion" ( and also above )

Reply:
> also typo "ion"
:smile: :roll_eyes: 

I'll fix it using your suggestion. Thanks.